src/Room.ts

```diff
@@ -64,14 +65,21 @@ export class Room<State= any> {
         this.connection = new Connection();
         this.connection.events.onmessage = this.onMessageCallback.bind(this);
         this.connection.events.onclose = (e: CloseEvent) => {
-            if (!this.hasJoined) {
-                console.warn(`Room connection was closed unexpectedly (${e.code}): ${e.reason}`);
-                this.onError.invoke(e.code, e.reason);
-                return;
-            }
+            if (devMode) {
+                setInterval(() => {
+                    this.connection = new Connection();
+                    this.connection.connect(endpoint);
+                }, 1000);
+            } else {
+                if (!this.hasJoined) {
+                    console.warn(`Room connection was closed unexpectedly (${e.code}): ${e.reason}`);
+                    this.onError.invoke(e.code, e.reason);
+                    return;
+                }
```

Comment:
If `this.hasJoined` is false there's no reason to even try to reconnect in case it's devMode. Can move this check to the very beginning and return early if this is false